commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/gitlab/GitlabPublisher.java

```diff
@@ -98,6 +107,28 @@ private void publish(@NotNull SBuild build,
       throw new PublisherException("Cannot parse repository URL from VCS root " + root.getName());
 
     String message = createMessage(status, build, revision, myLinks.getViewResultsUrl(build), description);
+
+    Pattern r = Pattern.compile("/merge-requests/([0-9]+)/");
+    Matcher m = r.matcher(revision.getRepositoryVersion().getVcsBranch());
+    if (m.find( )) {
+      String mr_number = m.group(1);
+
+      String url = GitlabSettings.getProjectsUrl(getApiUrl(), repository.owner(), repository.repositoryName()) + "/merge_requests/" + mr_number;
+
+      MRInfoResponseProcessor processorMR = new MRInfoResponseProcessor(); 
+
+      try {
+        HttpHelper.get(url, null, null, Collections.singletonMap("PRIVATE-TOKEN", getPrivateToken()), BaseCommitStatusPublisher.DEFAULT_CONNECTION_TIMEOUT, processorMR);
```

Comment:
The HTTP request is made synchronously within the CommitStatusPublisherListener event handlers execution. It may cause the control not to be returned to the code thatr calls these event handlers promptly enough, which is not desirable.